nuxeo-services/nuxeo-platform-convert/src/main/java/org/nuxeo/ecm/platform/convert/plugins/UTF8CharsetConverter.java

```diff
@@ -95,18 +95,31 @@ protected Blob convert(Blob blob) throws IOException, ConversionException {
         return newBlob;
     }
 
+    /**
+     * @deprecated since 11.1. Please use instead {@link UTF8CharsetConverter#detectEncoding(Blob)}
+     */
+    @Deprecated
     protected String detectEncoding(InputStream in) throws IOException, ConversionException {
-        if (!in.markSupported()) {
+        return getEncoding(in).orElseThrow(() -> new ConversionException("Cannot detect source charset."));
+    }
+
+    protected String detectEncoding(Blob blob) throws IOException, ConversionException {
+        try (InputStream stream = blob.getStream()) {
+            return getEncoding(stream).orElseThrow(
+                    () -> new ConversionException("Cannot detect source charset.", blob));
+        }
+    }
+
+    private Optional<String> getEncoding(InputStream in) throws IOException {
+        InputStream inputStream = Objects.requireNonNull(in);
```

Comment:
I'm not sure this one is really needed. If `null`, it will throw the same NPE in the next line...